### PR TITLE
[POAE7-2553] String Scalar Function support: length

### PR DESCRIPTION
**Pull request by YBRua**
### What changes were proposed in this pull request?

Added support for string scalar function `LENGTH(str)`, which computes the number of characters in an input `str`.

Notes:

1. There is an unused class `CharLengthExpr : Expr` in the code base (`Analyzer.h`), but to maintain consistency with existing string operations, this class is *not* used and a new `CharLengthStringOper : StringOper` is used instead.
2. The formal name for this function according to the substrait specification should be `char_length`, but DuckDb only supports `LENGTH`, and therefore all test cases used `LENGTH` for compatibility with duckdb.
3. Isthumus does not support converting `LENGTH()` or `CHAR_LENGTH()`. Current test cases are generated by a modified local fork of Isthmus.

### Why are the changes needed?

String op support.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UTs.

### CODE REVIEW

File: cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp [hunk 1]
```diff
@@ -37,8 +37,15 @@ bool getExprUpdatable(std::unordered_map<std::shared_ptr<Analyzer::Expr>, bool>
 }
 
 bool isStringFunction(const std::string& function_name) {
-  std::unordered_set<std::string> supportedStrFunctionSet{
-      "substring", "substr", "lower", "upper", "trim", "ltrim", "rtrim"};
+  std::unordered_set<std::string> supportedStrFunctionSet{"substring",
```
Comment (harborn) [inline cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp:40]:
```c++
static std::unordered_set<std::string> funcs = {"substring", ...};
```
1. declare it as a static variable
2. naming the variable as short as it.

File: cider/function/string/StringOps.cpp [hunk 2]
```diff
@@ -100,6 +100,16 @@ Datum TryStringCast::numericEval(const std::string_view str) const {
   }
 }
 
+NullableStrType CharLength::operator()(const std::string& str) const {
+  CIDER_THROW(CiderCompileException, "invalid call to operator() for CharLength");
+}
+
+Datum CharLength::numericEval(const std::string_view str) const {
```
Comment (harborn) [inline cider/function/string/StringOps.cpp:107]:
```c++
const std::string_view& str
```

File: cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp (hunk 1 above)
Comment (YBRua) [reply cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp]: updated

File: cider/function/string/StringOps.cpp (hunk 2 above)
Comment (YBRua) [reply cider/function/string/StringOps.cpp]: updated

**Review by jikunshang — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**